src/layout.rs

```diff
@@ -288,171 +277,142 @@ fn split(area: Rect, layout: &Layout) -> Rc<[Rect]> {
-                        elements[i].height
-                            | EQ(MEDIUM)
-                            | (f64::from(dest_area.height) * f64::from(n) / f64::from(d))
-                    }
-                    Constraint::Min(v) => elements[i].height | GE(MEDIUM) | f64::from(v),
-                    Constraint::Max(v) => elements[i].height | LE(MEDIUM) | f64::from(v),
-                });
-
-                match *size {
-                    Constraint::Min(v) | Constraint::Max(v) => {
-                        ccs.push(elements[i].height | EQ(WEAK) | f64::from(v));
-                    }
-                    _ => {}
-                }
-            }
-        }
-    }
-    solver.add_constraints(&ccs).unwrap();
-    for &(var, value) in solver.fetch_changes() {
-        let (index, attr) = vars[&var];
-        let value = if value.is_sign_negative() {
-            0
-        } else {
-            value as u16
-        };
-        match attr {
-            0 => {
-                results[index].x = value;
-            }
-            1 => {
-                results[index].y = value;
-            }
-            2 => {
-                results[index].width = value;
-            }
-            3 => {
-                results[index].height = value;
-            }
-            _ => {}
-        }
-    }
+    {
+        use taffy::prelude::*;
+        let mut taffy = Taffy::with_capacity(layout.constraints.len());
+        let mut taffy_keys = Vec::with_capacity(layout.constraints.len());
+
+        // disabling rounding, because otherwise the following tests fail:
+        // widgets_table_columns_widths_can_use_mixed_constraints
+        // widgets_table_column_spacing_can_be_changed
+        //
+        // but disabling rounding makes the following tests fail:
+        // layout::tests::test_vertical_split_by_height
+        taffy.disable_rounding();
+
+        for elt in &layout.constraints {
+            // alias for dest_area width or height, depending on layout.direction
+            let dest_area_var = match layout.direction {
+                Direction::Horizontal => dest_area.width,
+                Direction::Vertical => dest_area.height,
+            };
 
-    if layout.expand_to_fill {
-        // Fix imprecision by extending the last item a bit if necessary
-        if let Some(last) = results.last_mut() {
-            match layout.direction {
-                Direction::Vertical => {
-                    last.height = dest_area.bottom() - last.y;
-                }
-                Direction::Horizontal => {
-                    last.width = dest_area.right() - last.x;
-                }
-            }
-        }
-    }
-    res
-}
+            // value to use for the dimension (special case for percent)
+            let elt_applied = match elt {
+                // special case for percent, because it could be directly supported by taffy
+                // Constraint::Percentage(v) => (*v as f32) / 100.0,
```

Comment:
it should also maybe be discussed which to use: Constraint Percent implementation, or taffy's

Reply:
true, but wouldnt this also be somewhat of a breaking change? not using the `apply` method would make it obsolete, but removing it would be breaking because it is public API

Reply:
We can just not use the ignore the apply method (the taffy size stuff is more powerful anyway), then mark it as deprecated and remove it at some point in the future.

Reply:
`apply` has been deprecated with 8cdfe2a6b16cdfc6255cd4f42ac2b3027d6f0b32